ledger/src/blockstore_processor.rs

```diff
@@ -1410,26 +1403,43 @@ fn load_frozen_forks(
 
     let on_halt_store_hash_raw_data_for_debug = opts.on_halt_store_hash_raw_data_for_debug;
     if Some(bank_forks.read().unwrap().root()) != opts.halt_at_slot {
+        let recyclers = VerifyRecyclers::default();
+        let mut all_banks = HashMap::new();
+
+        const STATUS_REPORT_INTERVAL: Duration = Duration::from_secs(2);
+        let mut last_status_report = Instant::now();
+        let mut slots_processed = 0;
+        let mut txs = 0;
         let mut set_root_us = 0;
         let mut root_retain_us = 0;
         let mut process_single_slot_us = 0;
         let mut voting_us = 0;
+
         while !pending_slots.is_empty() {
             timing.details.per_program_timings.clear();
             let (meta, bank, last_entry_hash) = pending_slots.pop().unwrap();
             let slot = bank.slot();
-            if last_status_report.elapsed() > Duration::from_secs(2) {
+            if last_status_report.elapsed() > STATUS_REPORT_INTERVAL {
```

Comment:
out of scope/follow-up: seems like this is essentially an inlined `report` of what might be better off as a Metrics/Stats wrapper struct that publishes datapoints?

Reply:
Kind of,
- I could see throwing everything into a struct to reinforce grouping / move the logic to reset counters to a sub-function (ie `.reset()` after every update).
- However, I don't think we need to submit metrics for these intermediate updates. `load_frozen_forks()` returns the total time spent and the number of banks processed, so you can determine the average. My thought process here is you would look at the total time spent on metrics, and if that looks off, you go digging in logs to see intermediate and debug from there.